### DISCUSSION TITLE
orderBy relation average (or workaround)

### DISCUSSION THREAD

**Discussion by adrienharnay** [prisma/prisma#13245] [Q&A]
Hi!

I am currently trying to order `Games` based on `GameRatings` and am struggling to see how to do it with Prisma.

My model:

```prisma
model Game {
  id            String                 @id @default(uuid())
  playerRatings GameRating[]
}

model GameRating {
  id      String   @id @default(uuid())
  score   Float
  gameId  String
}
```

My query:

```ts
prisma.game.findMany({
  where: { ... },
  orderBy: {}, // ???
});
```

What I would like to do is something like this:

```ts
  orderBy: { playerRatings: { score: { _avg: true, groupBy: ['gameId'] } } }
```

I have read the docs, and I understand this is not possible with Prisma.

I have managed to get the data pretty easily with a separate query:

```ts
const res = await prisma.gameRating.groupBy({
  by: ['gameId'],
  _avg: { score: true },
});
```

which yields

```ts
[
  {
    _avg: { score: 5 },
    gameId: 'e79882a0-8af5-432b-aed8-6c70bff38f02'
  }
]
```

But then I don't know how to combine the two queries (average and list all games - with support for pagination, I can't get all the games and then order them based on the first query).

I am fairly new to Prisma and relational databases, so I don't know what options to explore. Would it make sense to:
- create a temporary table (?) with the `Games` data + the average and then query it with `orderBy: { computedAvg: 'desc' }`?
- create a column in the `Games` table and update the `computedAvg` value with a hourly/daily CRON job?

Again, if there were a way to combine the query I would prefer this option, but please don't hesitate to point me to a workaround if no other options are available.

Thanks in advance!

**Comment by pantharshit00** [comment]
Hello @adrienharnay , our apologies for the delay in getting back to you! Due to the large number of questions in our community channels, unfortunately this slipped through the cracks. It's very important for us to address your question, so let us know in a comment if this is still an open discussion. 

If this is still relevant, will the following query work for you:
```ts
  const res = await prisma.gameRating.groupBy({
    by: ["gameId"],
    _avg: { score: true },
    orderBy: {
      _avg: {
        score: "desc",
      },
    },
  });
```

**Reply by pantharshit00** [reply]
Hello @adrienharnay,

I am going to through past discussion again right now and saw this is still open. Can you please confirm if this is still relevant to you?

**Reply by adrienharnay** [reply]
Hello, thank you for your answer! I have just seen it, I am sorry for the delay.

I think your solution returns `gameRatings`, but I want to query `games` (ordered by gameRatings). Please let me know if I missed something!

**Comment by adeelbaig625** [comment]
@adrienharnay is there any solution?

**Reply by adrienharnay** [reply]
Not that I am aware of.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
